feg/gateway/services/session_proxy/servicers/credits.go

```diff
@@ -92,20 +92,18 @@ func makeCCRInit(
 	} else {
 		glog.Warning("No RatSpecificContext is specified")
 	}
-	request.Credits = makeUsedCreditsForCCRInit(keys)
+	request.Credits = makeUsedCreditsForCCRInit(pReq.RequestedUnits, keys)
 	return request
 }
 
-func makeUsedCreditsForCCRInit(keys []policydb.ChargingKey) []*gy.UsedCredits {
+func makeUsedCreditsForCCRInit(
+	requestedUnits *protos.RequestedUnits,
+	keys []policydb.ChargingKey) []*gy.UsedCredits {
 	usedCredits := make([]*gy.UsedCredits, 0, len(keys))
 	for _, key := range keys {
-		// TODO: move this into configuration on sessiond
-		// Default Requested-Service-Unit value for CCR-I
-		defaultRSU := &protos.RequestedUnits{Total: 10000, Tx: 10000, Rx: 10000}
-
 		uc := &gy.UsedCredits{
 			RatingGroup:    key.RatingGroup,
-			RequestedUnits: defaultRSU,
+			RequestedUnits: requestedUnits,
```

Comment:
if requestedUnits == nil { requestedUnits = defaultRSU }

Reply:
Yes, i had some checks before but I removed them thinking that it would be defaulted.... but i may be wrong.

So then, if the client uses an older proto which doesn't include the new field, will not the server fill that in with a default value?

Reply:
yea, default value for proto struct will be nil pointer